src/contracts/pods/EigenPod.sol

```diff
@@ -164,267 +175,355 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
     }
 
     /**
-     * @notice  This function verifies that the withdrawal credentials of the podOwner are pointed to
+     * @notice  This function verifies that the withdrawal credentials for one or many validators of the podOwner that are pointed to
      * this contract. It also verifies the current (not effective) balance  of the validator.  It verifies the provided proof of the ETH validator against the beacon chain state
      * root, marks the validator as 'active' in EigenLayer, and credits the restaked ETH in Eigenlayer.
-     * @param oracleBlockNumber is the Beacon Chain blockNumber whose state root the `proof` will be proven against.
-     * @param validatorIndex is the index of the validator being proven, refer to consensus specs 
-     * @param proofs is the bytes that prove the ETH validator's balance and withdrawal credentials against a beacon chain state root
+     * @param oracleTimestamp is the Beacon Chain timestamp whose state root the `proof` will be proven against.
+     * @param validatorIndices is a list of indices of the validators being proven, refer to consensus specs 
+     * @param proofs is the bytes that prove the ETH validator's  withdrawal credentials against a beacon chain state root
      * @param validatorFields are the fields of the "Validator Container", refer to consensus specs 
      * for details: https://github.com/ethereum/consensus-specs/blob/dev/specs/phase0/beacon-chain.md#validator
      */
-    function verifyWithdrawalCredentialsAndBalance(
-        uint64 oracleBlockNumber,
-        uint40 validatorIndex,
-        BeaconChainProofs.ValidatorFieldsAndBalanceProofs calldata proofs,
-        bytes32[] calldata validatorFields
-    )
-        external
+    function verifyWithdrawalCredentials(
+        uint64 oracleTimestamp,
+        uint40[] calldata validatorIndices,
+        BeaconChainProofs.WithdrawalCredentialProofs[] calldata proofs,
+        bytes32[][] calldata validatorFields
+    ) external 
+        onlyEigenPodOwner
         onlyWhenNotPaused(PAUSED_EIGENPODS_VERIFY_CREDENTIALS)
-        // check that the provided `oracleBlockNumber` is after the `mostRecentWithdrawalBlockNumber`
-        proofIsForValidBlockNumber(oracleBlockNumber)
+        // check that the provided `oracleTimestamp` is after the `mostRecentWithdrawalTimestamp`
+        proofIsForValidTimestamp(oracleTimestamp)
+        //ensure that caller has restaking enabled by calling "activateRestaking()"
```

Comment:
documentation is misleading. function is permissionless, what matters is the *pod* (not the *caller*) having restaking enabled.
Also current wording makes me think this modifier calls `activateRestaking` but rather it is that the pod owner must have previously called the function

Reply:
This function is permissioned to the EigenPodOwner

Reply:
> This function is permissioned to the EigenPodOwner (line 194)

Oh, hmm OK then. Any specific reasoning behind this change? I thought we had agreed the key shift was requiring the pod owner to explicitly call the `activateRestaking` function

Reply:
I think its still a good idea to permission verifyWithdrawalCredentials to prevent the griefing vector.  Even if you've activated restaking, makes sense taht you are the only one that can actually restake/lockup your funds.

Reply:
> I think its still a good idea to permission verifyWithdrawalCredentials to prevent the griefing vector. Even if you've activated restaking, makes sense taht you are the only one that can actually restake/lockup your funds.

does it though? once you've activated restaking, isn't the only way to withdraw to first `verifyWithdrawalCredentials`?

I guess this could help to a degree w/ the difficulty we were discussing about interactions with the delegation process.
I feel OK about the change for now, but think it's something we may revisit.